File: testdata/SampleSolution/SampleApp.Api/Services/ConfigConsumer.cs
using Microsoft.Extensions.Configuration;

namespace SampleApp.Api.Services;

public class ConfigConsumer
{
    private readonly IConfiguration _configuration;

    public ConfigConsumer(IConfiguration configuration)
    {
        _configuration = configuration;
    }

    public string GetConnectionString()
    {
        return _configuration["ConnectionStrings:DefaultDB"]!;
    }

    public int GetMaxRetries()
    {
        return _configuration.GetValue<int>("App:MaxRetries");
    }

    public string GetLogLevel()
    {
        return _configuration.GetSection("Logging:LogLevel:Default").Value!;
    }
}

File: testdata/SampleVbSolution/SampleVbApp/Api/ConfigConsumer.vb
Imports SampleVbApp.Stubs

Namespace Api

    ''' <summary>Demonstrates IConfiguration usage patterns for VB.NET.</summary>
    Public Class ConfigConsumer

        Private ReadOnly _config As IConfiguration

        Public Sub New(config As IConfiguration)
            _config = config
        End Sub

        Public Function GetConnectionString() As String
            Return _config.GetItem("ConnectionStrings:Default")
        End Function

        Public Function GetMaxRetries() As Integer
            Return _config.GetValue(Of Integer)("App:MaxRetries")
        End Function

        Public Function GetFeatureSection() As IConfiguration
            Return _config.GetSection("Features")
        End Function

    End Class

End Namespace
